<h1>The Dark Side of Big Data</h1>

In [2]:
print("Hello World.")

Hello World.


In [11]:
 import pandas as pd

In [12]:
df = pd.read_csv("Adult")

In [13]:
df.head()

age         workclass  demogweight  education  education-num  \
0   39         State-gov        77516  Bachelors             13   
1   50  Self-emp-not-inc        83311  Bachelors             13   
2   38           Private       215646    HS-grad              9   
3   53           Private       234721       11th              7   
4   28           Private       338409  Bachelors             13   

       marital-status         occupation   relationship   race     sex  \
0       Never-married       Adm-clerical  Not-in-family  White    Male   
1  Married-civ-spouse    Exec-managerial        Husband  White    Male   
2            Divorced  Handlers-cleaners  Not-in-family  White    Male   
3  Married-civ-spouse  Handlers-cleaners        Husband  Black    Male   
4  Married-civ-spouse     Prof-specialty           Wife  Black  Female   

   capital-gain  capital-loss  hours-per-week native-country  income  
0          2174             0              40  United-States  <=50K.  
1             0             0              13  United-States  <=50K.  
2             0             0              40  United-States  <=50K.  
3             0             0              40  United-States  <=50K.  
4             0             0              40           Cuba  <=50K.

In [15]:
df.shape

(25000, 15)

<p>25000 is the number of rows. 15 is the number of columns.</p>

In [16]:
df = df.rename(columns={'education-num': 'educationnum','marital-status': 'marital','capital-gain':'capitalgain','capital-loss':'capitalloss','hours-per-week':'hoursperweek','native-country':'nativecountry'})

In [17]:
df.head()

age         workclass  demogweight  education  educationnum  \
0   39         State-gov        77516  Bachelors            13   
1   50  Self-emp-not-inc        83311  Bachelors            13   
2   38           Private       215646    HS-grad             9   
3   53           Private       234721       11th             7   
4   28           Private       338409  Bachelors            13   

              marital         occupation   relationship   race     sex  \
0       Never-married       Adm-clerical  Not-in-family  White    Male   
1  Married-civ-spouse    Exec-managerial        Husband  White    Male   
2            Divorced  Handlers-cleaners  Not-in-family  White    Male   
3  Married-civ-spouse  Handlers-cleaners        Husband  Black    Male   
4  Married-civ-spouse     Prof-specialty           Wife  Black  Female   

   capitalgain  capitalloss  hoursperweek  nativecountry  income  
0         2174            0            40  United-States  <=50K.  
1            0            0            13  United-States  <=50K.  
2            0            0            40  United-States  <=50K.  
3            0            0            40  United-States  <=50K.  
4            0            0            40           Cuba  <=50K.

In [18]:
df.race.unique()

array(['White', 'Black', 'Asian-Pac-Islander', 'Amer-Indian-Eskimo',
       'Other'], dtype=object)

In [19]:
df.nativecountry.unique()

array(['United-States', 'Cuba', 'Jamaica', 'India', '?', 'Mexico',
       'South', 'Puerto-Rico', 'Honduras', 'England', 'Canada', 'Germany',
       'Iran', 'Philippines', 'Italy', 'Poland', 'Columbia', 'Cambodia',
       'Thailand', 'Ecuador', 'Laos', 'Taiwan', 'Haiti', 'Portugal',
       'Dominican-Republic', 'El-Salvador', 'France', 'Guatemala',
       'China', 'Japan', 'Yugoslavia', 'Peru',
       'Outlying-US(Guam-USVI-etc)', 'Scotland', 'Trinadad&Tobago',
       'Greece', 'Nicaragua', 'Vietnam', 'Hong', 'Ireland', 'Hungary',
       'Holand-Netherlands'], dtype=object)

In [20]:
df.describe()

age   demogweight  educationnum   capitalgain  capitalloss  \
count  25000.000000  2.500000e+04  25000.000000  25000.000000  25000.00000   
mean      38.606920  1.897418e+05     10.081440   1088.579040     86.49992   
std       13.687769  1.052947e+05      2.557035   7486.621197    401.25358   
min       17.000000  1.228500e+04      1.000000      0.000000      0.00000   
25%       28.000000  1.179630e+05      9.000000      0.000000      0.00000   
50%       37.000000  1.783530e+05     10.000000      0.000000      0.00000   
75%       48.000000  2.368610e+05     12.000000      0.000000      0.00000   
max       90.000000  1.484705e+06     16.000000  99999.000000   4356.00000   

       hoursperweek  
count  25000.000000  
mean      40.407440  
std       12.299267  
min        1.000000  
25%       40.000000  
50%       40.000000  
75%       45.000000  
max       99.000000

<p>a. 38.6</p>
<p>b. 90</p>
<p>c. 10. I would interpret it as 50 percent recieve less than 10 in education and other 50 percent recieve more than 10.</p>
<p>d. Mean</p>
<p>e. The mean shows it to be an average of approx. 1088.6 per individual but the median shows 50 percent and below recieve 0 capital gain</p>